src/Aspire.Hosting/ApplicationModel/ParameterResource.cs

```diff
@@ -25,6 +24,7 @@ public ParameterResource(string name, Func<ParameterDefault?, string> callback,
         ArgumentNullException.ThrowIfNull(callback);
 
         _valueGetter = callback;
+        _lazyValue = new Lazy<string>(() => _valueGetter(Default));
```

Comment:
What is the default safety mode when not specified explicitly? Is it thread safe?

I think it would be clearer to specify it explicitly.

Reply:
ExecutionAndPublication, the one that you always want by default 😄